include/swift/SIL/SILInstruction.h

```diff
@@ -637,6 +807,9 @@ class ApplyInstBase<Impl, Base, false> : public Base {
   /// list's tail allocation.
   unsigned NumSubstitutions;
 
+  /// The number of call arguments as required by the callee.
+  unsigned NumCallArguments;
+
   /// Used for apply_inst instructions: true if the called function has an
   /// error result but is not actually throwing.
   bool NonThrowing;
```

Comment:
You could pack NonThrowing with NumSubstitutions and avoid growing the size of an ApplyInst.  2 billion substitutions is probably enough, even for the standard library. :)